src/modules/purchase/usePurchase.ts

```diff
@@ -0,0 +1,90 @@
+import { Purchase } from "@/domain/model/purchase/Purchase";
+import {
+  useMutation,
+  useQueryClient,
+  useSuspenseQuery,
+} from "@tanstack/react-query";
+import { PurchaseService } from "@/domain/model/purchase/PurchaseService";
+import { getPurchaseAPIClient } from "@/external-interfaces/api";
+
+export function useQueryPurchase({ id }: { id: Purchase["id"] }) {
+  const purchaseService = new PurchaseService(getPurchaseAPIClient());
+
+  return useSuspenseQuery({
+    queryKey: ["purchase", id],
+    queryFn: () => {
+      return purchaseService.getPurchaseItem(id);
+    },
+  });
+}
+
+export function useQueryPurchases() {
+  const purchaseService = new PurchaseService(getPurchaseAPIClient());
+
+  return useSuspenseQuery({
+    queryKey: ["purchases"],
+    queryFn: () => {
+      return purchaseService.getPurchaseList();
+    },
+  });
+}
+
+function useRefetchPurchases() {
+  const queryClient = useQueryClient();
+
+  return () => {
+    return queryClient.refetchQueries({ queryKey: ["purchases"] });
+  };
+}
+
+function useRefetchPurchase() {
+  const queryClient = useQueryClient();
+
+  return (id: string) => {
+    return queryClient.refetchQueries({ queryKey: ["purchase", id] });
+  };
+}
+
+export function useCreatePurchase() {
+  const purchaseService = new PurchaseService(getPurchaseAPIClient());
+  const refetchPurchases = useRefetchPurchases();
+  const refetchPurchase = useRefetchPurchase();
+
+  return useMutation({
+    mutationFn: (
+      ...params: Parameters<typeof purchaseService.createPurchase>
+    ) => purchaseService.createPurchase(...params),
+    onSuccess: ({ id }) => {
+      return Promise.all([refetchPurchases(), refetchPurchase(id)]);
+    },
```

Comment:
생각해보니까 요 mutation은 create라 refetchPurchase가 없어도 될 것 같네요? 이거는 나중에 다른곳이랑 일괄로 삭제하시죠~
```suggestion
    onSuccess: () => {
      return refetchPurchases()
    },
```